panther_lights_scheduler/src/scheduler_node.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+import rospy
+
+from sensor_msgs.msg import BatteryState
+from std_msgs.msg import Bool
+
+from panther_msgs.msg import LEDAnimation
+from panther_msgs.srv import SetLEDAnimation, SetLEDAnimationRequest
+
+
+class LightsSchedulerNode:
+    def __init__(self, name) -> None:
+        rospy.init_node(name, anonymous=False)
+
+        self._battery_percentage = 1.0
+        self._charger_connected = False
+        self._e_stop_state = None
+        self._led_e_stop_state = None
+        self._last_battery_anim_time = 0.0
+
+        self._critical_battery_anim_period = rospy.get_param('critical_battery_anim_period', 15.0)
+        self._critical_battery_margin = rospy.get_param('critical_battery_margin', 0.1)
+        self._low_battery_anim_period = rospy.get_param('low_battery_anim_period', 30.0)
+        self._low_battery_margin = rospy.get_param('low_battery_margin', 0.4)
+
+        # -------------------------------
+        #   Publishers & Subscribers
+        # -------------------------------
+
+        rospy.Subscriber('battery', BatteryState, self._battery_cb, queue_size=1)
+        rospy.Subscriber(
+            'hardware/charger_connected', Bool, self._charger_connected_cb, queue_size=1
+        )
+        rospy.Subscriber('hardware/e_stop', Bool, self._e_stop_cb, queue_size=1)
+
+        # -------------------------------
+        #   Services
+        # -------------------------------
+
+        self._set_led_animation = rospy.ServiceProxy(
+            '/lights/controller/set/animation', SetLEDAnimation
+        )
+
+        # -------------------------------
+        #   Timers
+        # -------------------------------
+
+        self._scheduler_timer = rospy.Timer(rospy.Duration(0.1), self._scheduler_timer_cb)
+
+        rospy.loginfo(f'{rospy.get_name()} Node started')
+
+    def _scheduler_timer_cb(self, *args) -> None:
+
+        if self._led_e_stop_state != self._e_stop_state:
+            req = SetLEDAnimationRequest()
+            req.repeating = True
+            if self._e_stop_state:
+                req.animation.id = LEDAnimation.E_STOP
+                req.animation.name = 'E_STOP'
```

Comment:
If we are calling animation by both ID and its name, I would add the string to be by default stored as a constant in a message or got rid of the `name` filed from message definition

Reply:
Having constants limits you to basic animations, I left is as variable so user can define own animations and still specify name.

Reply:
I still have to be convinced about it